### ISSUE TITLE
2.1.24 - Parsing error when adding password with two underscore characters

### ISSUE THREAD

**Issue by hexxamillion** [nilsteampassnet/TeamPass#1382]
We get this error if a password being entered has two underscore characters in it. Where is the parsing for passwords handled. You should be able to put whatever in a password. It may be possible this is an issue on new versions as well. How can I fix on 2.1.24?

An error appears. Answer from Server cannot be parsed!
Returned data:
Error: 
Query: UPDATE `cpassman_items` SET `label`='TestItem', `description`='', `pw`='75d227a99b1cce6c84954410dbadc2ee', `pw_iv`='2152528f081318e9a1343097eba79565', `email`='', `login`='blah', `url`='', `id_tree`='467', `restricted_to`='', `anyone_can_modify`='1', `complexity_level`='100' WHERE id=1230

**Comment by nilsteampassnet** [comment]
I've tried with last version
![image](https://cloud.githubusercontent.com/assets/1197546/16901293/72212380-4c3f-11e6-9991-16bdcf93bf78.png)
But I don't see why it should not work with 2 underscore characters, this seems really strange.

I've tried the query in phpmyadmin and it didn't generate any error message.
Have you tried this? Just to get the error message.

**Comment by hexxamillion** [comment]
I see now. It's really odd behavior. It's almost as if it happens after a certain number of characters. Maybe it has nothing to do with underscores.

I can get the error if I use this: abcde_fg_hi-jkl12nt
I do not if I use this: abcde_fg_hi-

To rule out issue with underscores I tested with this and still got the error: abcde1fg3hi*jkl12nt
 I am wondering now if it has to do with some limitation on password length.

**Comment by hexxamillion** [comment]
Just following up on this. I found that the error occurs only after a certain number of characters. See my previous note. Possible limitation on password length.

**Comment by hexxamillion** [comment]
I wanted to follow up on this again.

**Comment by hexxamillion** [comment]
I am wondering if there is a solution to this. I just had another occurrence today. Can you help me understand where the password length is dictated in the code? I validated that if they shortened the password from 15 to 12 characters it worked successfully.

**Comment by hexxamillion** [comment]
I found the admin setting for max password length but it seems the setting is not being applied. For example I can set and save the password length of 20. I see the update in the database but when you go to try it I still get the error above unless I use a password that is only 12 characters. Any suggestions?

**Comment by hexxamillion** [comment]
I found out how to fix this.

In the logs I saw this which is strange and I never would have thought. In fact, I am not even sure how this relates.

INSERT INTO `cpassman_items` (`label`,`description`,`pw`,`pw_iv`,`email`,`url`,`id_tree`,`login`,`inactif`,`restricted_to`,`perso`,`anyone_can_modify`,`complexity_level`) VALUES ('test000', '', '2cba4abe45d56d1916b9a0b144602f85', '3dfece4d1d088982df347f01a7d31113', '', '', '7', 'test', '0', '', '0', '1', '100')<br />Data too long for column 'complexity_level' at row 1@/sources/items.queries.php

What I did was alert the column complexity_level varchar length from 2  to 25 in the cpassman_items table. Then after that I was able to successfully enter long passwords. Strange bug.
